zipline/data/minute_bars.py

```diff
@@ -108,6 +109,46 @@ def _sid_subdir_path(sid):
     )
 
 
+def convert_col(col, scale_factor, sid, col_name, invalid_data_behavior):
+    """Adapt float column into a uint32 column.
+
+    Parameters
+    ----------
+    col : ndarray
+        Float column to convert to uint32.
+    scale_factor : int
+        Factor to use to scale float values before converting to uint32.
+    sid : int
+        Sid of the relevant asset, for logging.
+    col_name : str
+        Name of the columns, for logging.
+    invalid_data_behavior : str
+        Specifies behavior when data cannot be converted to uint32.
+        If 'raise', raises an exception.
+        If 'warn', logs a warning and filters out incompatible values.
+        If 'ignore', silently filters out incompatible values.
+    """
+    scaled_col = np.nan_to_num(col) * scale_factor
+    max_val = scaled_col.max()
+
+    try:
+        check_uint32_safe(max_val, col_name)
+    except ValueError:
+        if invalid_data_behavior == 'raise':
+            raise
+
+        if invalid_data_behavior == 'warn':
+            logger.warn(
+                'Values for sid={}, col={} contain some too large for '
+                'uint32 (max={}), filtering them out',
+                sid, col_name, max_val,
+            )
+
+        return scaled_col[scaled_col <= np.iinfo(np.uint32).max]
```

Comment:
This will make the results no longer align, because elements which do not meet the predicate will be removed.

e.g.:
```
In [37]: foo = np.random.rand(10)

In [38]: foo
Out[38]: 
array([ 0.86608194,  0.03375079,  0.76099617,  0.29876178,  0.01236926,
        0.17863316,  0.85661549,  0.96673179,  0.12907978,  0.11848878])

In [39]: foo[foo < 0.5]
Out[39]: 
array([ 0.03375079,  0.29876178,  0.01236926,  0.17863316,  0.12907978,
        0.11848878])
```